docs/_docs/03-ug-mutable-dataset.md

```diff
@@ -199,35 +197,35 @@ query.show
 #### Append and Delete dataset
 
 Now, we can consider handling deleted files. Basically, Hybrid Scan excludes indexed data from deleted source files
-by scanning all index rows and verifying whether each is coming from a deleted source file or not.
+by scanning all index rows and verifying whether each row is coming from a deleted source file or not.
 In order to trace which source file each row is from, you need to enable linage column config before creating an index.
-Check the [configuration](https://microsoft.github.io/hyperspace/docs/ug-configuration/) page to see how lineage is
-enabled when creating an index.
+Check [configuration](https://microsoft.github.io/hyperspace/docs/ug-configuration/) page to see how to enable the lineage column at index creation.
 
 Due to the way Hybrid Scan enforces deletes at the query time, supporting deletes is more expensive than appended
-files. The more deleted files it has, the more overhead it will incur to filter the rows.
+files. The more deleted files it has, the more overhead it will incur to filter the rows, and the benefit from the index will decrease.
 Therefore, you need to be aware of possible performance regression from it.
 
-We will provide several threshold configs after some experiments and optimizations.
-
 ###### How to enable
 
 You can use the following configurations to enable Hybrid Scan for indexes on a dataset with both append and delete files.
 
-We currently provide one threshold config:
-`spark.hyperspace.index.hybridscan.delete.maxNumDeleted`. If there are more deleted files than the config value,
-we do not perform Hybrid Scan for the index.
+We currently provide one threshold config for deleted files:
+`spark.hyperspace.index.hybridscan.maxDeletedRatio`
+
+It indicates the maximum ratio of deleted data compared to index data to perform Hybrid Scan.
```

Comment:
throughout the doc could you please add the info what is the ration about?
`maximum ratio of deleted data compared to index data`
Is it data file size? Is it num files?